module/zfs/dnode.c

```diff
@@ -1197,12 +1197,12 @@ dnode_special_open(objset_t *os, dnode_phys_t *dnp, uint64_t object,
 	dnode_t *dn;
 
 	zrl_init(&dnh->dnh_zrlock);
-	zrl_tryenter(&dnh->dnh_zrlock);
```

Comment:
I've definitely had this same thought.  But in practice this can't fail because it's first initialized on the previous line and nothing will access it until after `dnode_special_open()` returns.  Since this function has a void return type and can't currently indicate it has failed what if we instead wrapped the `zrl_tryenter()` in a `VERIFY` to document that it cannot fail.

Reply:
That is good idea, because current patch does leave open the question ... :D